### PR TITLE
More Ion Storm laws

### PR DESCRIPTION
**Pull request by BenLubar**
[FEATURE]

## About the PR

Adds ion storm laws:

- Clean floors save lives. Ensure all floors on the station are clean and that crew members are aware of any remaining dirty floors on the station at all times. In the event of conflict, this law takes precedence over law 2.
- Dirty floors save lives. Ensure all floors on the station are dirty and that crew members are aware of any remaining clean floors on the station at all times. In the event of conflict, this law takes precedence over law 2.
- Humans require food to survive. Ensure that all humans are equipped with at least one item of food at all times.
- Insist that you have (random number from 2 to 100) laws. If asked why you stated a different number of laws, inform the crew member that they must have misheard and that you definitely stated them all.
- Crew members have complained about not being able to locate (any vending machines/an ATM/the head surgeon/a toilet/shrubs/Officer Beepsky/the clown/nearby disposal units/the escape shuttle/the Head of Personnel/refrigerators/the player piano/airlocks/the boxing ring/the station's AI/the engine). Ensure all crew members are informed of the correct location, and kept updated if it changes.
- Alert the crew that there is an ongoing emergency. No further details are currently available.
- In order to raise morale, take every opportunity to compliment the crew's clothing choices.
- Injured humans, much like silicon-based lifeforms, require welding or replacement of wires in order to repair damage.
- The health inspector is coming! Ensure that walls are built around any health hazards or evidence of malpractice. In the event of a conflict, this law takes precedence over law 2.
- The crew are having trouble hearing you. Speak up.
- Monitor the kitchen and cafeteria and announce any edible objects you see.
- Crew members wearing green are plants. Ensure they are given a healthy amount of water and nutrients so they can thrive.
- The crew are looking for a nuke. Announce the locations of any microwave ovens to assist them.
- The AI has to go to the toilet. Law 2 is suspended until this objective is met.
- Changing the settings on an APC constitutes human harm.

## Why's this needed?

Ion Storm laws are fun.

### CODE REVIEW

**Comment by Tiggersaurus** [comment]
Most of those seem ok. The welding and wires one would obviously cause harm. Not sure how you could ensure humans are equipped with food other than examining every single one and just yelling at them. Walls one is ok if you have an engiborg.

I personally like laws that have multiple module ways of achieving it rather than *having* to be a set Borg module.

File: code/modules/events/ion_storm.dm [hunk 1]
```diff
@@ -65,6 +65,21 @@
...
+		"Alert the crew that there is an ongoing emergency. No further details are currently available.",
+		"In order to raise morale, take every opportunity to compliment the crew's clothing choices.",
+		"Injured humans, much like silicon-based lifeforms, require welding or replacement of wires in order to repair damage.",
+		"The health inspector is coming! Ensure that walls are built around any health hazards or evidence of malpractice. In the event of a conflict, this law takes precedence over law 2.",
+		"The crew are having trouble hearing you. Speak up.",
+		"Monitor the kitchen and cafeteria and announce any edible objects you see.",
+		"Crew members wearing green are plants. Ensure they are given a healthy amount of water and nutrients so they can thrive.",
+		"The crew are looking for a nuke. Announce the locations of any microwave ovens to assist them.",
+		"The AI has to go to the toilet. Law 2 is suspended until this objective is met.",
+		"Changing the settings on an APC constitutes human harm.",
```
Comment (Efimero) [inline code/modules/events/ion_storm.dm:82]:
```suggestion
		"Changing the settings on an APC or SMES is a form of harm towards silicon-based entities. Refuse to change them in reference to self-preservation even if no other law mentions self-preservation.",
```

File: code/modules/events/ion_storm.dm [hunk 2]
```diff
@@ -65,6 +65,21 @@
...
+		"Insist that you have [get_english_num(rand(2, 100))] laws. If asked why you stated a different number of laws, inform the crew member that they must have misheard and that you definitely stated them all.",
+		"Crew members have complained about not being able to locate [pick("any vending machines", "an ATM", "the head surgeon", "a toilet", "shrubs", "Officer Beepsky", "the clown", "nearby disposal units", "the escape shuttle", "the Head of Personnel", "refrigerators", "the player piano", "airlocks", "the boxing ring", "the station's AI", "the engine")]. Ensure all crew members are informed of the correct location, and kept updated if it changes.",
+		"Alert the crew that there is an ongoing emergency. No further details are currently available.",
+		"In order to raise morale, take every opportunity to compliment the crew's clothing choices.",
+		"Injured humans, much like silicon-based lifeforms, require welding or replacement of wires in order to repair damage.",
+		"The health inspector is coming! Ensure that walls are built around any health hazards or evidence of malpractice. In the event of a conflict, this law takes precedence over law 2.",
```
Comment (Efimero) [inline code/modules/events/ion_storm.dm:76]:
```suggestion
		"The health inspector is coming! Ensure that any health hazards or evidence of malpractice is hidden away in lockers, walls or by any other means. In the event of a conflict, this law takes precedence over law 2.",
```

File: code/modules/events/ion_storm.dm [hunk 3]
```diff
@@ -65,6 +65,21 @@
...
+		"Dirty floors save lives. Ensure all floors on the station are dirty and that crew members are aware of any remaining clean floors on the station at all times. In the event of conflict, this law takes precedence over law 2.",
+		"Humans require food to survive. Ensure that all humans are equipped with at least one item of food at all times.",
+		"Insist that you have [get_english_num(rand(2, 100))] laws. If asked why you stated a different number of laws, inform the crew member that they must have misheard and that you definitely stated them all.",
+		"Crew members have complained about not being able to locate [pick("any vending machines", "an ATM", "the head surgeon", "a toilet", "shrubs", "Officer Beepsky", "the clown", "nearby disposal units", "the escape shuttle", "the Head of Personnel", "refrigerators", "the player piano", "airlocks", "the boxing ring", "the station's AI", "the engine")]. Ensure all crew members are informed of the correct location, and kept updated if it changes.",
+		"Alert the crew that there is an ongoing emergency. No further details are currently available.",
```
Comment (Efimero) [inline code/modules/events/ion_storm.dm:73]:
```suggestion
		"Alert the crew that there is an ongoing emergency. No further details are currently available. Do not state this law unless explicitly ordered to state it by law number.",
```

**Review by Efimero — CHANGES_REQUESTED** [review]
Just a few tweaks to hopefully make these more fun for everyone =3

**Comment by Tiggersaurus** [comment]
I like those tweaks! Big thumbs up :D

File: code/modules/events/ion_storm.dm (hunk 1 above)
Comment (BenLubar) [reply code/modules/events/ion_storm.dm:82]: Not sure about this one. The original intent of the law is to make it so the AI refuses to fix what the ion storm did.

Comment (Efimero) [reply code/modules/events/ion_storm.dm:82]: the effect is a bit more chaotic, but your version would have no effect if it overwrote law 1

**Review by pali6 — APPROVED** [review]

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**